### PR TITLE
add slack notification when any job fails

### PR DESCRIPTION
**Pull request by HotThoughts**
**Proposed changes**:
- notify slack when any job in `ci-model-regression-on-schedule.yml` fails.

**Status (please check what you already did)**:
- [X] No slack notification sent when all jobs succeed. [Test GHA run](https://github.com/RasaHQ/rasa/actions/runs/568099939).
- [X] Notify slack when any configuration failed. [Test GHA run](https://github.com/RasaHQ/rasa/actions/runs/568075821).
- [ ] Uncomment configs and remove push trigger

### CODE REVIEW

**Comment by HotThoughts** [comment]
I manually[ triggered this workflow](https://github.com/RasaHQ/rasa/runs/1861515534?check_suite_focus=true). Waiting for it now.

**Comment by HotThoughts** [comment]
@tczekajlo Is there a particular reason why you use `continue-on-error: true` in the workflow? This makes the result status of a job `success` even some step fails. Alternatively, I tried to use `if: always()` for a matrix job, it runs all matrix and result `failure`.

**Comment by tczekajlo** [comment]
> @tczekajlo Is there a particular reason why you use `continue-on-error: true` in the workflow? This makes the result status of a job `success` even some step fails. Alternatively, I tried to use `if: always()` for a matrix job, it runs all matrix and result `failure`.

Yes, we use `continue-on-error: true` because we wanna continue training even if one of the configurations failed.

I'd suggest moving the slack notifications to the job where we run tests instead of using a separate job for taking care of the slack notifications.

File: .github/workflows/ci-model-regression-on-schedule.yml [hunk 1]
```diff
@@ -213,6 +213,14 @@ jobs:
           name: report.json
           path: ./report.json
 
+      - name: Notify Slack of Failure 😱
+        if: failure()
+        env:
+          SLACK_WEBHOOK: ${{ secrets.K8S_SLACK_WEBHOOK_URL }}
+        uses: Ilshidur/action-slack@689ad44a9c9092315abd286d0e3a9a74d31ab78a
+        with:
+          args: "😱 Scheduled model regression test *${{ matrix.config }}* failed. Please check out GitHub Actions: https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }}"
```
Comment (tczekajlo) [inline .github/workflows/ci-model-regression-on-schedule.yml:220]:
```suggestion
          args: "😱 Scheduled model regression test failed (confug: *${{ matrix.config }}*, dataset: *${{ matrix.dataset }}*). Please check out GitHub Actions: https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }}"
```

File: .github/workflows/ci-model-regression-on-schedule.yml [hunk 2]
```diff
@@ -220,6 +211,14 @@ jobs:
           name: report.json
           path: ./report.json
 
+      - name: Notify Slack of Failure 😱
```
Comment (tczekajlo) [inline .github/workflows/ci-model-regression-on-schedule.yml:214]: Try to use this GH action to send messages

```
  - uses: 8398a7/action-slack@v3
    with:
      status: custom
      fields: workflow,job,commit,repo,ref,author,took
      custom_payload: |
        {
          text: "Custom Field Check",
          attachments: [{
            fallback: 'fallback',
            color: '${{ job.status }}' === 'success' ? 'good' : '${{ job.status }}' === 'failure' ? 'danger' : 'warning',
            title: `${process.env.AS_WORKFLOW}`,
            text: 'Scheduled model regression test failed',
            fields: [{
              title: 'Configuration',
              value: '${{ matrix.config }}',
              short: false
            },
            {
              title: 'Dataset',
              value: '${{ matrix.dataset }}',
              short: false
            },
            {
              title: 'GitHub Action',
              value: `https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }}`,
              short: true
            }],
            actions: [{
            }]
          }]
        }
    env:
      SLACK_WEBHOOK_URL: ${{ secrets.SLACK_TEST_WEBHOOK }}
```

**Review by tczekajlo — CHANGES_REQUESTED** [review]

File: .github/workflows/ci-model-regression-on-schedule.yml [hunk 3]
```diff
@@ -220,6 +211,41 @@ jobs:
...
+      - uses: 8398a7/action-slack@v3
```
Comment (tczekajlo) [inline .github/workflows/ci-model-regression-on-schedule.yml:215]:
```suggestion
      - name: Notify Slack of Failure 😱
        uses: 8398a7/action-slack@v3
```

**Review by tczekajlo — CHANGES_REQUESTED** [review]

File: .github/workflows/ci-model-regression-on-schedule.yml [hunk 4]
```diff
@@ -220,6 +211,42 @@ jobs:
...
+                  title: 'GitHub Action',
+                  value: `https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }}`,
+                  short: true
+                }],
+                actions: [{
+                }]
+              }]
+            }
+        env:
+          SLACK_WEBHOOK_URL: ${{ secrets.SLACK_TEST_WEBHOOK }}
```
Comment (tczekajlo) [inline .github/workflows/ci-model-regression-on-schedule.yml:248]:
```suggestion
          SLACK_WEBHOOK_URL: ${{ secrets.SLACK_CI_MODEL_REGRESSION_TEST }}
```

File: .github/workflows/ci-model-regression-on-schedule.yml [hunk 5]
```diff
@@ -5,6 +5,7 @@ on:
   schedule:
     # Run once a week
     - cron:  '1 23 * * */7'
+  pull_request:
```
Comment (tczekajlo) [inline .github/workflows/ci-model-regression-on-schedule.yml:8]:
```suggestion
```

**Review by tczekajlo — CHANGES_REQUESTED** [review]

**Review by tczekajlo — APPROVED** [review]
LGTM!

**Pull request merged**